### PR TITLE
[scintilla] Add static linking to Scintilla

### PR DESCRIPTION
**Pull request by alek-d**
**Describe the pull request**

- What does your PR fix? Fixes #

Static linking is officially supported by Scintilla. See https://www.scintilla.org/ScintillaDoc.html, at the bottom of the document, section "Building Scintilla". This PR adds static linking on Windows.

- Which triplets are supported/not supported? Have you updated the CI baseline?

Tested successfully:
x64-windows
x64-windows-static
x64-windows-static-md
x86-windows
x86-windows-static
x86-windows-static-md

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes, it does.

### CODE REVIEW

**Comment by cenit** [comment]
you should not call `vcpkg_apply_patches` in a portfile.
Please define a symbol referring to the patch file for static or empty for dynamic and then add the `PATCHES` section to `vcpkg_extract_source_archive_ex`

File: ports/scintilla/0001-static-build.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+       <LanguageStandard>stdcpp17</LanguageStandard>
++      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
+     </ClCompile>
+     <Link>
+       <EnableCOMDATFolding>true</EnableCOMDATFolding>
+@@ -129,6 +132,7 @@
+       <IntrinsicFunctions>true</IntrinsicFunctions>
+       <PreprocessorDefinitions>NDEBUG;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+       <LanguageStandard>stdcpp17</LanguageStandard>
++      <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
```
Comment (Hoikas) [inline ports/scintilla/0001-static-build.patch:42]: These changes to the RuntimeLibrary force usage of the static CRT. They should be pulled into a separate patch, conditioned on `VCPKG_CRT_LINKAGE`.

File: ports/scintilla/portfile.cmake [hunk 2]
```diff
@@ -13,10 +11,20 @@ vcpkg_extract_source_archive_ex(
...
+
 vcpkg_install_msbuild(
   SOURCE_PATH ${SOURCE_PATH}
   PROJECT_SUBPATH Win32/SciLexer.vcxproj
   INCLUDES_SUBPATH include
   LICENSE_SUBPATH License.txt
   ALLOW_ROOT_INCLUDES
 )
+
+vcpkg_copy_pdbs()
```
Comment (Hoikas) [inline ports/scintilla/portfile.cmake:30]: Please include a trailing newline.

**Review by Hoikas — CHANGES_REQUESTED** [review]

File: ports/scintilla/portfile.cmake [hunk 3]
```diff
@@ -13,10 +11,20 @@ vcpkg_extract_source_archive_ex(
   REF 4.2.3
 )
```
Comment (ras0219-msft) [inline ports/scintilla/portfile.cmake:12]:
```suggestion
  PATCHES ${PATCHES}
)
```
You can then conditionally set `PATCHES` based on `VCPKG_CRT_LINKAGE`:
```cmake
if(VCPKG_CRT_LINKAGE STREQUAL "static")
  set(PATCHES 0001-static-build.patch)
else()
  set(PATCHES)
endif()
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/scintilla/portfile.cmake (hunk 3 above)
Comment (alek-d) [reply ports/scintilla/portfile.cmake:22]: Thank you for the tip.

File: ports/scintilla/portfile.cmake (hunk 2 above)
Comment (alek-d) [reply ports/scintilla/portfile.cmake]: Included.

File: ports/scintilla/0001-static-build.patch (hunk 1 above)
Comment (alek-d) [reply ports/scintilla/0001-static-build.patch]: Good point. I assumed that static triplets always links CRT statically.
I've separated these two options, and now CRT can be linked dynamically.
Tested with two community triplets `x64/x86-windows-static-md` – works fine.
Thanks.

Comment (Hoikas) [reply ports/scintilla/0001-static-build.patch]: Thanks! :smile:

**Review by PhoebeHui — APPROVED** [review]

**Comment by cenit** [comment]
`scintilla:x64-windows-static=fail` in `ci.baseline.txt`, but in reality is also described as skip

```
            scintilla:x64-windows-static:       skip: 3e9793440929e680a883b0c2d392ba56798acdbb
```

this PR should fix this problem and let CI test the port maybe?

**Comment by alek-d** [comment]
@cenit Thanks. `ci.baseline.txt` updated. 

I have a question about other triplets. Maybe all other currently marked as `fail` should be marked as `skip`. 
Current Scintilla `portfile.cmake` start with `vcpkg_fail_port_install(ON_TARGET "Linux" "OSX" "UWP")`. 

So, maybe it would be better to just skip those build rather than expecting them to fail?
Should I change to this?
```
# Currently Scintilla build only on Windows
scintilla:arm-uwp=skip
scintilla:x64-linux=skip
scintilla:x64-osx=skip
scintilla:x64-uwp=skip
```

**Comment by cenit** [comment]
No leave them as fail.
Skip is only for ports conflicting with each other (name conflicts in header or lib filenames)

**Comment by alek-d** [comment]
Removed `vcpkg_copy_pdbs()` from `portfile.cmake` because it is called in `vcpkg_install_msbuild()`.

File: ports/scintilla/portfile.cmake (hunk 3 above)
Comment (alek-d) [reply ports/scintilla/portfile.cmake:22]: @ras0219-msft could you please approve the change. I would like to update Scintilla to the newest version in another PR, but this one is hanging for so long.

**Comment by ras0219-msft** [comment]
LGTM, sorry for the long delay on this PR and thanks for addressing my comment!
